api/src/main/java/gov/cms/ab2d/api/controller/BulkDataAccessAPI.java

```diff
@@ -109,11 +120,33 @@
     )
     @GetMapping(value = "/Job/{jobId}/$status")
     @ResponseStatus(value = HttpStatus.OK)
-    public ResponseEntity<Void> getJobStatus(
-            @ApiParam(value = "A job identifier", required = true) @PathVariable @NotBlank String jobId) throws IOException {
+    public ResponseEntity<JsonNode> getJobStatus(
+            @ApiParam(value = "A job identifier", required = true) @PathVariable @NotBlank String jobId) {
+        Job job = jobService.getJobByJobID(jobId);
+
+        LocalDateTime now = LocalDateTime.now();
+
+        if (job.getLastPollTime() != null && job.getLastPollTime().plusSeconds(retryAfterDelay).isAfter(now)) {
+            throw new TooManyRequestsException("You are polling too frequently");
+        }
+
+        job.setLastPollTime(now);
+        jobService.updateJob(job);
+
         HttpHeaders responseHeaders = new HttpHeaders();
-        responseHeaders.add("X-Progress", "0%");
-        return new ResponseEntity<>(null, responseHeaders, HttpStatus.ACCEPTED);
+        if (job.getProgress().equals(100)) {
+            responseHeaders.add("Expires", DateUtil.formatLocalDateTimeAsUTC(job.getExpires()));
+            JobCompletedResponse resp = new JobCompletedResponse();
+            resp.setTransactionTime(new DateTimeType(convertLocalDateTimeToDate(job.getCompletedAt())).toHumanDisplay());
+            resp.setRequest(job.getRequestURL());
+            resp.setOutput(job.getJobOutput().stream().filter(o -> !o.isError()).map(o -> new JobCompletedResponse.Output(o.getFhirResourceType(), o.getFilePath())).collect(Collectors.toList()));
+            resp.setError(job.getJobOutput().stream().filter(o -> o.isError()).map(o -> new JobCompletedResponse.Output(o.getFhirResourceType(), o.getFilePath())).collect(Collectors.toList()));
+            return new ResponseEntity<>(new ObjectMapper().valueToTree(resp), responseHeaders, HttpStatus.OK);
+        } else {
```

Comment:
We should return this _only_ if the job is in `SUBMITTED` or `IN_PROGRESS` status